ChangeLog.d/remove_allow_sha1_in_certificates

```diff
@@ -0,0 +1,15 @@
+Removals
+   * Remove optional SHA-1 in the default TLS configuration for certificate
+     signing. This feature was ment to be available only temporarily.
+     Users are expected to use SHA-2 instead, since SHA-1 is currently
+     considered a security risk.
```

Comment:
The first sentence is hard to understand. What we're doing here is removing the compile-time option that changes the default behavior of TLS connections to allow SHA-1-signed certificates. It is not removing the permission to use SHA-1 by default: that was off by default a long time ago. Suggested rewording:

> Remove the compile-time option MBEDTLS_TLS_DEFAULT_ALLOW_SHA1_IN_CERTIFICATES, which was off by default. Users should not trust certificates signed with SHA-1 due to the known attacks against SHA-1.

(I think the name of the option is clear enough not to require additional explanation.)